components/reddit/sources/new-comments-by-user/new-comments-by-user.js

```diff
@@ -0,0 +1,78 @@
+const common = require("../common");
+const { reddit } = common.props;
+
+module.exports = {
+  ...common,
+  key: "new-comments-by-user",
+  name: "New comments by user",
+  description: "Emits an event each time a user posts a new comment.",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    ...common.props,
+    username: { propDefinition: [common.props.reddit, "username"] },
+    numberOfParents: {
+      type: "integer",
+      label: "Number of parents",
+      description:
+        "The emitted events will contain the new comment plus the parents of said comment up to the number indicated in this property.",
+      default: 2,
+      min: 2,
+      max: 10,
+    },
+    timeFilter: { propDefinition: [common.props.reddit, "timeFilter"] },
+    includeSubredditDetails: {
+      propDefinition: [reddit, "includeSubredditDetails"],
+    },
+  },
+  hooks: {
+    async deploy() {
+      // Emits sample events on the first run during deploy.
+      var redditComments = await this.reddit.getNewUserComments(
+        null,
+        this.username,
+        this.numberOfParents,
+        this.timeFilter,
+        this.includeSubredditDetails,
+        10
+      );
+      const { children: comments = [] } = redditComments.data;
+      if (comments.length === 0) {
+        console.log("No data available, skipping itieration");
+        return;
+      }
+      const { name: before = this.db.get("before") } = comments[0].data;
+      this.db.set("before", before);
+      comments.reverse().forEach(this.emitRedditEvent);
+    },
+  },
+  methods: {
+    ...common.methods,
+    generateEventMetadata(redditEvent) {
+      return {
+        id: redditEvent.data.name,
+        summary: redditEvent.data.body,
+        ts: redditEvent.data.created,
+      };
+    },
+  },
+  async run() {
+    do {
+      const redditComments = await this.reddit.getNewUserComments(
+        this.db.get("before"),
+        this.username,
+        this.numberOfParents,
+        this.timeFilter,
+        this.includeSubredditDetails
+      );
+      const { children: comments = [] } = redditComments.data;
+      if (comments.length === 0) {
+        console.log("No data available, skipping itieration");
+        break;
+      }
+      const { name: before = this.db.get("before") } = comments[0].data;
+      this.db.set("before", before);
+      comments.reverse().forEach(this.emitRedditEvent);
+    } while (redditComments);
```

Comment:
This won't paginate because `redditComments` will be undefined in this context, since you set it _within_ the loop. You'll need to declare it at the top of the `run` method, outside the loop:

```
let redditComments;
```

then remove the `const` in front of `redditComments` within the loop.

Reply:
My bad. I know this upfront, of the control variable out of scope in the while statement being defined inside the loop, but lost focus when cleaning up the code as per other comments in the review.
I made this changes in a couple of other sources where this was needed.